src/tokens/TokenImport.tsx

```diff
@@ -27,57 +30,149 @@ import variables from 'src/styles/variables'
 import { PasteButton } from 'src/tokens/PasteButton'
 import { tokensByIdSelector } from 'src/tokens/selectors'
 import { getTokenId } from 'src/tokens/utils'
-import networkConfig from 'src/web3/networkConfig'
-import { isAddress } from 'viem'
+import Logger from 'src/utils/Logger'
+import { publicClient } from 'src/viem'
+import networkConfig, { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { Address, formatUnits, getContract, isAddress } from 'viem'
+
+const TAG = 'tokens/TokenImport'
 
 type Props = NativeStackScreenProps<StackParamList, Screens.TokenImport>
 
+const FETCH_TOKEN_DETAILS_TIMEOUT_MS = 5_000
```

Comment:
i've heard that RPC / general network requests are significantly slower in some regions (some folks mentioned this after a research trip to Nigeria) - i think 5 seconds may be too short, could we increase this? maybe can use the existing `FETCH_TIMEOUT_DURATION` instead

Reply:
I've changed to using Viem's timeout. As we use the default HTTP transport, it means 10 second timeout. We can change that later in the future in `src/viem/index.ts`.